prow/scripts/build-kcp-cli.sh

```diff
@@ -58,16 +59,22 @@ gcp::authenticate \
 log::info "Content of the local artifacts directory"
 ls -la "${ARTIFACTS}"
 
-log::info "Copy artifacts to ${KCP_DEVELOPMENT_ARTIFACTS_BUCKET}/${BUCKET_DIR}"
-
-gsutil cp "${ARTIFACTS}/kcp.exe" "${KCP_DEVELOPMENT_ARTIFACTS_BUCKET}/${BUCKET_DIR}/kcp.exe"
-gsutil cp "${ARTIFACTS}/kcp-linux" "${KCP_DEVELOPMENT_ARTIFACTS_BUCKET}/${BUCKET_DIR}/kcp-linux"
-gsutil cp "${ARTIFACTS}/kcp-darwin" "${KCP_DEVELOPMENT_ARTIFACTS_BUCKET}/${BUCKET_DIR}/kcp-darwin"
+for tool in {kcp,ers}; do
+    bucket=$(tool_bucket ${tool})
+    log::info "Copy artifacts to ${bucket}"
+    for ext in {.exe,-linux,-darwin}; do
+        file=${tool}${ext}
+        gsutil cp "${ARTIFACTS}/${file}" "${bucket}/${file}"
+    done
+done
```

Comment:
I don't understand why do you include some sophisticated logic here. I believe you want to copy the entire contents of `$ARTIFACTS` directory to the GCS bucket. To put it simply, make sure this directory contains all the files required and use only one command:
```
gsutil cp "$ARTIFACTS/*" gs://bucket
```
Do you really need to separate both files into different subdirectories?

Reply:
> Do you really need to separate both files into different subdirectories?

the copy directory has been defined in https://github.com/kyma-project/test-infra/pull/2972 as

https://github.com/kyma-project/test-infra/blob/e6322c1ea111408d9a51aee01afa9d67156acdeb/prow/scripts/build-kcp-cli.sh#L70-L72

which then evaluates to for example
```
gs://kyma-development-artifacts/kcp/master-57ac8a24/kcp-linux
gs://kyma-development-artifacts/kcp/master-57ac8a24/kcp-darwin
gs://kyma-development-artifacts/kcp/master-57ac8a24/kcp.exe
```
and these are distributed via the following links
```
https://storage.googleapis.com/kyma-development-artifacts/kcp/master/kcp-linux
https://storage.googleapis.com/kyma-development-artifacts/kcp/master/kcp-darwin
https://storage.googleapis.com/kyma-development-artifacts/kcp/master/kcp.exe
```

I don't have any intentions of redesigning the storage layout for kyma developer tools, it predates me but also it seems flexible enough, sufficient and most importantly, others are already used to it.

> I don't understand why do you include some sophisticated logic here

would you rather see this implemented as 6 independent copy calls than a for-loop? I'm ok either way, just my developer instincts kicked in and I thought it's less error-prone if it's written as a loop rather than copypasta.
```
gsutil cp "${ARTIFACTS}/kcp.exe" "${KCP_DEVELOPMENT_ARTIFACTS_BUCKET}/master/kcp.exe"
gsutil cp "${ARTIFACTS}/kcp-linux" "${KCP_DEVELOPMENT_ARTIFACTS_BUCKET}/master/kcp-linux"
gsutil cp "${ARTIFACTS}/kcp-darwin" "${KCP_DEVELOPMENT_ARTIFACTS_BUCKET}/master/kcp-darwin"
gsutil cp "${ARTIFACTS}/ers.exe" "${ERS_DEVELOPMENT_ARTIFACTS_BUCKET}/master/ers.exe" 
gsutil cp "${ARTIFACTS}/ers-linux" "${ERS_DEVELOPMENT_ARTIFACTS_BUCKET}/master/ers-linux" 
gsutil cp "${ARTIFACTS}/ers-darwin" "${ERS_DEVELOPMENT_ARTIFACTS_BUCKET}/master/ers-darwin"
```

Reply:
I replaced this by constructing proper directory structure on local filesystem and then `gsutil rsync` to the bucket. I think this is reasonably readable.

Reply:
I really like it!